### PR TITLE
Justin/APPEALS-8468 untie vlj

### PR DESCRIPTION
**Pull request by sulak-j**
Resolves [APPEALS-8468](https://vajira.max.gov/browse/APPEALS-8468)

### Description
- Verifying functionality works on cases being distributed regardless of whether they were tied to a VLJ or not behind acd_distribute_all feature toggle
- Setting the genpop parameter to only_genpop inside the hearing_request_docket as it only sees AMA appeals
- Set an early return statement within the hearing_request_distribution_query as both most_recent_held_hearings_not_tied_to_any_judge and most_recent_held_hearings_exceeding_affinity_threshold are redundant 

### Acceptance Criteria
- [ ] Code compiles correctly

* AMA Hearing docket cases are not only assigned to the VLJ who held the hearing. This currently occurs because of the change of hearing_case_affinity_days to 0 and should continue to occur.
For priority 
For non-priority
* Changes should be behind feature toggle acd_distribute_all 
Hearings Docket by default distributes as genpop when feature toggle is enabled
https://github.com/department-of-veterans-affairs/caseflow/blob/feature/APPEALS-8275/app/models/dockets/hearing_request_docket.rb#L43 
The parameter genpop is defaulted to "any" but only "genpop" will be occurring, a default value is used because at https://github.com/department-of-veterans-affairs/caseflow/blob/feature/APPEALS-8275/app/models/concerns/all_case_distribution.rb#L50 and https://github.com/department-of-veterans-affairs/caseflow/blob/feature/APPEALS-8275/app/models/concerns/all_case_distribution.rb#L59 the parameter is not passed in so a default is needed. Passing in a parameter of genpop would affect other docket's distribution so need to only update hearing_request_docket
* Hearings Docket queries are simplified to reflect that there are no more "tied to VLJ" cases in the docket
most_recent_held_hearings_not_tied_to_any_judge and most_recent_held_hearings_exceeding_affinity_threshold are now redundant as all Hearing Docket Cases are considered not "tied to a VLJ"
https://github.com/department-of-veterans-affairs/caseflow/blob/feature/APPEALS-8275/app/queries/hearing_request_distribution_query.rb#L38

### CODE REVIEW

File: app/models/distribution.rb [hunk 1]
```diff
@@ -115,9 +115,8 @@ def longer_than_thirty_days_ago(date)
   end
 
   def assigned_tasks
-    pending_statuses = [Constants.TASK_STATUSES.assigned, Constants.TASK_STATUSES.in_progress]
     judge.tasks.select do |t|
-      t.is_a?(JudgeAssignTask) && pending_statuses.include?(t.status)
+      t.is_a?(JudgeAssignTask) && t.active?
```
Comment (ThorntonMatthew) [inline app/models/distribution.rb:119]: `.active?` is located in the Task class here: https://github.com/department-of-veterans-affairs/caseflow/blob/6ecbf208935da0565281b98faea5df88bb504048/app/models/task.rb#L336

`active_statuses` just so happens to contain the same statuses that `pending_statuses` did:
```ruby
    def active_statuses
      [Constants.TASK_STATUSES.assigned, Constants.TASK_STATUSES.in_progress]
    end
```

File: app/queries/hearing_request_distribution_query.rb [hunk 2]
```diff
@@ -37,9 +37,14 @@ def not_genpop_appeals
   # due to incompatibilities between the two queries.
```
Comment (ThorntonMatthew) [inline app/queries/hearing_request_distribution_query.rb:37]: This comment is going to need to be updated.

File: app/queries/hearing_request_distribution_query.rb [hunk 3]
```diff
@@ -37,9 +37,14 @@ def not_genpop_appeals
...
     no_hearings_or_no_held_hearings = with_no_hearings.or(with_no_held_hearings)
+
+    # returning early as most_recent_held_hearings_not_tied_to_any_judge is redundant
+    return no_hearings_or_no_held_hearings if FeatureToggle.enabled?(
+      :acd_distribute_all, user: RequestStore.store[current_user]
+    )
+
     [
       most_recent_held_hearings_not_tied_to_any_judge,
-      most_recent_held_hearings_exceeding_affinity_threshold,
```
[the remark this answers is not in the document]
Comment (ThorntonMatthew) [reply app/queries/hearing_request_distribution_query.rb:42]: Adding back in `most_recent_held_hearings_exceeding_affinity_threshold` causes many of the tests in `spec/models/dockets/hearing_request_docket_spec.rb` that were failing without it to pass.

[the remark this answers is not in the document]
Comment (ThorntonMatthew) [reply app/queries/hearing_request_distribution_query.rb:42]: For AMA appeals, I would think `with_no_hearings.or(with_no_held_hearings)` would be sufficient in grabbing all desired appeals.. Are we missing out on any due to moving from `genpop: "any"` to `"genpop: only_genpop"`?

**Comment by ThorntonMatthew** [comment]
8468-results

Quite a few rspec failures are occurring. Many of them are feature tests where pages are [404](https://http.cat/404)'ing. I'll take a sample of some of the failures and run them locally in hopes that I can find the common deficiency in the code.

**Comment by ThorntonMatthew** [comment]
screenshot_2022-09-22-08-40-21 321

I noticed some seed updates within this feature work, and I'm wondering if tests are expecting data that is no longer there. Just a running theory.

File: app/queries/hearing_request_distribution_query.rb (hunk 3 above)
Comment (zurbergram) [inline app/queries/hearing_request_distribution_query.rb:42]: this shouldn't be removed for Proportions distribution, please add back `most_recent_held_hearings_exceeding_affinity_threshold`

**Review by zurbergram — CHANGES_REQUESTED** [review]
`app/queries/hearing_request_distribution_query.rb` needs to be changed.

Please update the requested method comment as well

**Review by zurbergram — APPROVED** [review]

**Comment by ThorntonMatthew** [comment]
```ruby
    context "nonpriority appeals and genpop 'any'" do
      before { FeatureToggle.enable!(:acd_distribute_all) }
      after { FeatureToggle.disable!(:acd_distribute_all) }
```

That'll do it! 😅

**Review by shruthisibi — APPROVED** [review]

**Comment by ThorntonMatthew** [comment]
The following appeal isn't showing up in the output of `HearingRequestDistributionQuery#only_genpop_appeals`: https://github.com/department-of-veterans-affairs/caseflow/blob/e7da7470def1305ea4c2e5108645e98a80f9ac80/spec/models/dockets/hearing_request_docket_spec.rb#L201

This is contributing to a test failure.

**Comment by ThorntonMatthew** [comment]
Treee for `inside_affinity`:
```ruby
                                                 ┌───────────────────────────────────────────────────────────────────────────┐
Appeal 8 (H 220921-8 Original) ───────────────── │ ID │ STATUS    │ ASGN_BY  │ ASGN_TO           │ UPDATED_AT                │
├── RootTask                                     │ 29 │ on_hold   │          │ Bva               │ 2022-09-22 11:14:02 -0400 │
│   └── DistributionTask                         │ 30 │ on_hold   │          │ Bva               │ 2022-09-22 11:14:02 -0400 │
│       └── HearingTask                          │ 31 │ completed │          │ Bva               │ 2022-09-22 11:14:02 -0400 │
│           ├── ScheduleHearingTask              │ 32 │ completed │          │ Bva               │ 2022-09-22 11:14:02 -0400 │
│           └── AssignHearingDispositionTask     │ 33 │ on_hold   │ CSS_ID19 │ Bva               │ 2022-09-22 11:14:02 -0400 │
│               ├── TranscriptionTask            │ 34 │ assigned  │          │ TranscriptionTeam │ 2022-09-22 11:14:02 -0400 │
│               └── EvidenceSubmissionWindowTask │ 35 │ assigned  │          │ MailTeam          │ 2022-09-22 11:14:02 -0400 │
└── EvidenceSubmissionWindowTask                 │ 36 │ completed │          │ MailTeam          │ 2022-09-22 11:14:02 -0400 │
                                                 └───────────────────────────────────────────────────────────────────────────┘
```

**Comment by ThorntonMatthew** [comment]
It looks like the method name (`create_nonpriority_unblocked_hearing_appeal_within_affinity`) may be a bit off given the affinity window change.

The crux of the issue appears to be that his method generates an appeal with a `DistributionTask` that has a status of `on_hold` instead of `assigned`, and that's not going to get picked up in this query:
```ruby
    def exceeding_affinity_threshold
      joins(with_assigned_distribution_task_sql)
        .where(hearings: { disposition: "held" })
        .where("distribution_task.assigned_at <= ?", Constants::DISTRIBUTION["hearing_case_affinity_days"].days.ago)
    end
```

For reference, here's `with_assigned_distribution_task_sql`:
```ruby
  def with_assigned_distribution_task_sql
    # both `appeal_type` and `appeal_id` necessary due to composite index
    <<~SQL
      INNER JOIN tasks AS distribution_task
      ON distribution_task.appeal_type = 'Appeal'
      AND distribution_task.appeal_id = appeals.id
      AND distribution_task.type = 'DistributionTask'
      AND distribution_task.status = 'assigned'
    SQL
  end
```

**Pull request merged**